mempool/cat/reactor.go

```diff
@@ -545,31 +564,34 @@ func (memR *Reactor) processPendingSeenForSigner(signer []byte) {
 		return
 	}
 
-	expectedSeq, haveExpected := memR.sequenceExpectationForSigner(signer, true, true)
+	expectedSeq, haveExpected := memR.querySequenceFromApplication(signer)
 	if !haveExpected {
+		memR.Logger.Error("no signer found in application")
 		return
 	}
 
 	for _, entry := range entries {
-		if memR.mempool.Has(entry.txKey) {
+		if haveExpected && expectedSeq > entry.sequence {
 			memR.pendingSeen.remove(entry.txKey)
 			continue
 		}
-		if memR.requests.ForTx(entry.txKey) != 0 {
+
+		if memR.mempool.Has(entry.txKey) {
 			memR.pendingSeen.remove(entry.txKey)
 			continue
 		}
-		if expectedSeq > entry.sequence {
-			memR.pendingSeen.remove(entry.txKey)
+
+		if memR.requests.ForTx(entry.txKey) != 0 || entry.requested {
```

Comment:
Can these be out of sync? i.e. the transaction has been requested but entry.requested is false? 

It would be simpler if we could safely rely on one variable